Sources/Stagehand/AnimationInstance/InteractiveAnimationInstance.swift

```diff
@@ -0,0 +1,223 @@
+            return
+        }
+
+        driver.animate(to: relativeTimestamp, using: curve, duration: duration, completion: completion)
+    }
+
+    public func animateToBeginning(
+        using curve: AnimationCurve = LinearAnimationCurve(),
+        duration: TimeInterval? = nil,
+        completion: ((_ finished: Bool) -> Void)? = nil
+    ) {
+        animate(to: 0, using: curve, duration: duration, completion: completion)
+    }
+
+    public func animateToEnd(
+        using curve: AnimationCurve = LinearAnimationCurve(),
+        duration: TimeInterval? = nil,
+        completion: ((_ finished: Bool) -> Void)? = nil
+    ) {
+        animate(to: 1, using: curve, duration: duration, completion: completion)
+    }
+
+    public func markAsComplete() {
+        driver.markAsComplete()
+        status = .complete
+    }
+
+    public enum CancelationBehavior {
+        /// Return the element back to its state at the beginning of the animation. This will cause the completion handlers to be called with a `finished` of false.
+        case revert
+        /// Stop the animation at its current progress. This will cause the completion handlers to be called with a `finished` of false.
+        case halt
+        /// Apply the final values of the animation. This will cause the completion handlers to be called with a `finished` of true.
+        case complete
+    }
+
+    public func cancel(behavior: CancelationBehavior = .halt) {
+        guard !status.isComplete else {
+            return
+        }
+
+        status = .canceled(behavior: behavior)
+        driver.animationInstanceDidCancel(behavior: behavior)
+    }
+
+    public enum Status {
+        case pending
+        case interactive(progress: Double)
+        case animating(progress: Double)
+        case complete
+        case canceled(behavior: CancelationBehavior)
+
+        var isComplete: Bool {
+            switch self {
+            case .pending, .interactive, .animating:
+                false
+            case .complete, .canceled:
+                true
+            }
+        }
+    }
+
+    public private(set) var status: Status = .pending
+
+    public var endToEndDuration: TimeInterval {
+        driver.endToEndDuration
+    }
+
+    // MARK: Internal
+
+    func executeBlocks(
+        from startingRelativeTimestamp: Double,
+        _ fromInclusivity: Executor.Inclusivity,
+        to endingRelativeTimestamp: Double
+    ) {
+        executor.executeBlocks(from: startingRelativeTimestamp, fromInclusivity, to: endingRelativeTimestamp)
+    }
+
+    /// Renders the frame at the specific timestamp, including rendering any keyframes between the timestamp between the
+    /// previously rendered frame and the specific timestamp.
+    ///
+    /// - parameter relativeTimestamp: The relative timestamp to render, with no curves applied.
+    func renderFrame(
+        at relativeTimestamp: Double
+    ) {
```

Comment:
are there any substantial differences between this method and the method with the same name on `AnimationInstance`? Without going line-by-line this looks like a copy/paste job. Probably fine, but might be worth calling out in a comment?

Reply:
Nah, it's effectively the same (just different status types). I'll leave a comment.